packages/aws/data_stream/cloudtrail/elasticsearch/ingest_pipeline/default.yml

```
@@ -51,13 +51,14 @@ processors:
   - script:
       description: Appends any relevant entity to `related.entity` for all events
       lang: painless
+      if: "ctx?.json?.eventName != null && ctx?.tags != null && ctx.tags.contains('actor_target_mapping')"
       on_failure: 
```

```suggestion
      on_failure: 
        - append:
            field: tags
            value: preserve_original_event
            allow_duplicates: false
```

Relates to https://github.com/elastic/integrations/issues/12045. This would allow for easier resolutions when things fail.

---

Why do we need the append if we would have added the tag `script_append_related_entity` already? (suggested in a previous comment)

---

Those are two different tag concepts. This comment is about the ECS `tags` field in the event. `script_append_related_entity` is about tagging the processor in the ingest pipeline (which shows up in [_nodes/stats](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-stats.html#cluster-nodes-stats-ingest-ex) and exceptions).

---

Check, I'm adding